File: src/Core/Domain/Aggregates/EmployeeAggregate/Employee.cs
using Application.Domain.Aggregates.AnimalOwnerAggregate;

namespace Application.Domain.Aggregates.EmployeeAggregate;

public class Employee : IEntity, IAggregateRoot
{
    public Employee()
    {
    }

    public Employee(string name, string surname, Role role, string address, string email, string password)
    {
        Name = name;
        Surname = surname;
        Role = role;
        Address = address;
        EmailAddress = new EmailAddress(email);
        Password = password;
    }

    public string Name { get; }
    public string Surname { get; }
    public Role Role { get; }
    public string Address { get; }
    public List<WorkSchedule> WorkSchedule { get; private set; }

    public EmailAddress EmailAddress { get; private set; }

    public string Password { get; private set; }
    public int Id { get; }
}

File: src/Client/src/types/employee.ts
import type Role from '@/enums/role.ts';

type Employee = {
    id: number;
    name: string;
    surname: string;
    email: string;
    role: Role;
    address: string;
};

type EmployeeCreateRequest = {
    name: string;
    surname: string;
    role: Role | null;
    address: string;
    email: string;
    password: string;
};

export type { Employee, EmployeeCreateRequest };
